src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -209,6 +209,12 @@ public void RemoveSourceFile(string filePath)
             _projectCodeModel.OnSourceFileRemoved(filePath);
         }
 
+        public void RenameSourceFile(string originalFilePath, string newFilePath)
+        {
+            _visualStudioProject.RenameSourceFile(originalFilePath, newFilePath);
+            _projectCodeModel.OnSourceFileRenaming(originalFilePath, newFilePath);
```

Comment:
It's not clear to me if this is safe to call on a background thread (I'm seeing evidence in both directions), but it's possible that having a proper workspace event means the code model can update when the event fires...?